datajunction-clients/python/datajunction/compile.py

```diff
@@ -680,40 +667,43 @@ def _deploy_nodes(
             )
             try:
                 rendered_node_config = deepcopy(node_config)
+                prefixed_name = f"{prefix}.{node_config.name}"
+                # pre-fix the query
                 if isinstance(
                     node_config.definition,
                     (TransformYAML, DimensionYAML, MetricYAML),
                 ):
                     rendered_node_config.definition.query = render_prefixes(
-                        node_config.definition.query,
+                        rendered_node_config.definition.query or "",
                         prefix,
                     )
+                # pre-fix the tags
```

Comment:
Is this essentially like a namespace for tags? I wonder if we should just explicitly give tags a similar kind of namespaced treatment

Reply:
Yes, that's right. Although on the 2nd look I think this code will prevent us from using any existing tags in these nodes.